package exception;

public class Ex2 {

	public static void main(String[] args) {
		/*
		 * finally 블록
		 * - 예외 발생 여부와 관계없이 무조건 수행해야하는 문장을 기술
		 * - 심지어 return 문을 만나 메서드를 종료하더라도
		 *   호출한 곳으로 돌아가기 전 finally 블록을 실행한 후 돌아감
		 */

		System.out.println("프로그램 시작!");
		
		try {
			String str ="";
			System.out.println(str.length()); // NullPointerException 예외
			
			System.out.println("try 블록 끝!");
		} catch(Exception e) {
			System.out.println("NullPointerException 예외 처리!");
		} finally {
			System.out.println("finally - 예외 발생 여부와 관계없이 실행됨!");
		}
		
		System.out.println("프로그램 종료!");
		
	}

}
